packages/nc-gui/components/dashboard/settings/DataSources.vue

```diff
@@ -0,0 +1,390 @@
+
+        $e('a:base:delete')
+
+        sources.splice(sources.indexOf(base), 1)
+        await loadProject()
+      } catch (e: any) {
+        message.error(await extractSdkResponseErrorMsg(e))
+      }
+    },
+  })
+}
+
+const toggleBase = async (base: BaseType, state: boolean) => {
+  try {
+    base.enabled = state
+    await $api.base.update(base.project_id as string, base.id as string, {
+      id: base.id,
+      project_id: base.project_id,
+      enabled: base.enabled,
+    })
+    await loadProject()
+  } catch (e: any) {
+    message.error(await extractSdkResponseErrorMsg(e))
+  }
+}
+
+const moveBase = async (e: any) => {
+  try {
+    if (e.oldIndex === e.newIndex) return
+    // sources list is mutated so we have to get the new index and mirror it to backend
+    const base = sources[e.newIndex]
+    if (base) {
+      if (!base.order) {
+        // empty update call to reorder bases (migration)
+        await $api.base.update(base.project_id as string, base.id as string, {
+          id: base.id,
+          project_id: base.project_id,
+        })
+        message.info('Bases are migrated. Please try again.')
+      } else {
+        await $api.base.update(base.project_id as string, base.id as string, {
+          id: base.id,
+          project_id: base.project_id,
+          order: e.newIndex + 1,
+        })
+      }
+    }
+    await loadProject()
+    await loadBases()
+  } catch (e: any) {
+    message.error(await extractSdkResponseErrorMsg(e))
+  }
+}
+
+onMounted(async () => {
+  if (sources.length === 0) {
+    await loadBases()
+  }
+})
+
+watch(
+  () => props.reload,
+  async (reload) => {
+    if (reload) {
+      await loadBases()
+    }
+  },
+)
+
+watch(
+  vState,
+  (newState) => {
+    switch (newState) {
+      case ClientType.MYSQL:
+        clientType = ClientType.MYSQL
+        vState.value = DataSourcesSubTab.New
+        break
+      case ClientType.PG:
+        clientType = ClientType.PG
+        vState.value = DataSourcesSubTab.New
+        break
+      case ClientType.SQLITE:
+        clientType = ClientType.SQLITE
+        vState.value = DataSourcesSubTab.New
+        break
+      case ClientType.MSSQL:
+        clientType = ClientType.MSSQL
+        vState.value = DataSourcesSubTab.New
+        break
+    }
+  },
+  { immediate: true },
+)
+</script>
+
+<template>
+  <div class="flex flex-row w-full">
+    <div class="flex flex-col w-full overflow-auto">
+      <div v-if="vState === ''" class="max-h-600px min-w-1200px overflow-y-auto">
+        <div class="ds-table-head">
+          <div class="ds-table-row">
+            <div class="ds-table-col ds-table-enabled"></div>
+            <div class="ds-table-col ds-table-name">Name</div>
+            <div class="ds-table-col ds-table-actions">Actions</div>
+          </div>
+        </div>
+        <div class="ds-table-body">
+          <Draggable :list="sources" item-key="id" handle=".ds-table-handle" @end="moveBase">
```

Comment:
Draggable icon is not visible in some screen resolution

<img width="183" alt="Screenshot 2022-10-31 at 2 37 56 PM" src="https://user-images.githubusercontent.com/61551451/198972436-4119ea8b-c9fc-43ab-8922-4c25396ef072.png">

Reply:
I separated show/hide icon to another column this also fixed issue with drag icon.